sanitizers/src/main/java/com/code_intelligence/jazzer/sanitizers/SqlInjection.kt

```diff
@@ -0,0 +1,165 @@
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.PreparedStatement",
+            targetMethod = "executeQuery"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.PreparedStatement",
+            targetMethod = "executeUpdate"
+        ),
+        MethodHook(type = HookType.REPLACE, targetClassName = "java.sql.CallableStatement", targetMethod = "execute"),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.CallableStatement",
+            targetMethod = "executeBatch"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.CallableStatement",
+            targetMethod = "executeLargeBatch"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.CallableStatement",
+            targetMethod = "executeLargeUpdate"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.CallableStatement",
+            targetMethod = "executeQuery"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "java.sql.CallableStatement",
+            targetMethod = "executeUpdate"
+        ),
+        MethodHook(
+            type = HookType.REPLACE,
+            targetClassName = "javax.persistence.EntityManager",
+            targetMethod = "createNativeQuery"
+        )
+    )
+    @JvmStatic
+    fun checkSqlExecute(method: MethodHandle, thisObject: Any?, arguments: Array<Any>, hookId: Int): Any {
+        var hasValidSqlQuery = false
+
+        if (arguments.isNotEmpty() && arguments[0] is String) {
+            val query = arguments[0] as String
+            hasValidSqlQuery = isValidSql(query)
+            Jazzer.guideTowardsContainment(query, CHARACTERS_TO_ESCAPE, hookId)
+        }
+        return try {
+            method.invokeWithArguments(thisObject, *arguments)
+        } catch (throwable: Throwable) {
+            // If we already validated the query string and know it's correct,
+            // The exception is likely thrown by a non-existent table or something
+            // that we don't want to report.
+            if (!hasValidSqlQuery && SQL_SYNTAX_ERROR_EXCEPTIONS.contains(throwable.javaClass.name)) {
+                Jazzer.reportFindingFromHook(
+                    FuzzerSecurityIssueHigh(
+                        """
+                    SQL Injection
+                    Injected query: ${arguments[0]}
+                        """.trimIndent(),
+                        throwable
+                    )
+                )
+            }
+            throw throwable
+        }
+    }
+
+    private fun isValidSql(sql: String): Boolean =
+        try {
+            CCJSqlParserUtil.parseStatements(sql)
+            true
+        } catch (e: JSQLParserException) {
```

Comment:
Should we add another catch block handling `Exception`, or even `Throwable`, to prevent exceptions thrown in the library from bubbling up into application code?

Reply:
Good point! I added a catch block that catches all `Throwable`.